src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -458,17 +458,13 @@ private IDataView GetShuffledData(string path)
             Tensor evaluationStep = null;
             Tensor prediction = null;
             Tensor bottleneckTensor = evalGraph.OperationByName(_bottleneckOperationName);
-
-            tf_with(evalGraph.as_default(), graph =>
-            {
-                var (_, _, groundTruthInput, finalTensor) = AddFinalRetrainOps(classCount, options.LabelColumn,
+            evalGraph.as_default();
```

Comment:
>evalGraph.as_default(); [](start = 12, length = 23)

Does this mutate `evalGraph`? In DnnUtils.cs line 96 you wrote `graph = graph.as_default();`.

Reply:
In line 842 in this file there is also a call to `as_default()` without assignment.

---
In reply to: [324450796](https://github.com/dotnet/machinelearning/pull/4205#discussion_r324450796) [](ancestors = 324450796)

Reply:
This sets the tensorflow default graph to the evalGraph. This does not mutate the evalGraph.

---
In reply to: [324450796](https://github.com/dotnet/machinelearning/pull/4205#discussion_r324450796) [](ancestors = 324450796)